crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -235,59 +235,12 @@ impl<T: Sync + Send + Clone, W: Sync + Send + Clone> EventStoring<ContractEvent>
     }
 
     async fn append_events(&mut self, events: Vec<EthContractEvent<ContractEvent>>) -> Result<()> {
-        let order_placement_events = events
-            .clone()
-            .into_iter()
-            .filter(|EthContractEvent { data, .. }| {
-                matches!(data, ContractEvent::OrderPlacement(_))
-            })
-            .collect();
-        let invalidation_events = get_invalidation_events(events)?;
-        let invalided_order_uids = extract_invalidated_order_uids(invalidation_events)?;
-        let (custom_order_data, quotes, broadcasted_order_data, orders) = self
-            .extract_custom_and_general_order_data(order_placement_events)
-            .await?;
-
-        let _timer = DatabaseMetrics::get()
-            .database_queries
-            .with_label_values(&["append_onchain_order_events"])
-            .start_timer();
-        let mut transaction = self.db.pool.begin().await?;
-
-        database::onchain_invalidations::insert_onchain_invalidations(
-            &mut transaction,
-            invalided_order_uids.as_slice(),
-        )
-        .await
-        .context("insert_onchain_invalidations failed")?;
-        database::onchain_broadcasted_orders::append(
-            &mut transaction,
-            broadcasted_order_data.as_slice(),
-        )
-        .await
-        .context("append_onchain_orders failed")?;
-
-        self.custom_onchain_data_parser
-            .append_custom_order_info_to_db(&mut transaction, custom_order_data)
-            .await
-            .context("append_custom_onchain_orders failed")?;
-
-        // We only need to insert quotes for orders that will be included in an
-        // auction (they are needed to compute solver rewards). If placement
-        // failed, then the quote is not needed.
-        insert_quotes(
-            &mut transaction,
-            quotes.into_iter().flatten().collect::<Vec<_>>().as_slice(),
+        // Appending events is equivalent to replacing events out of range.
+        self.replace_events(
```

Comment:
IMO, in order to make it fully modular, we should have two functions `delete_events()` and `append_events()` (and `replace_events()` calls them both). Conceptually we are calling a `replace_events()` to append events, but we just use a range out of bounds so it doesn't replace any events, but the intention is not to replace events.

It is more of a concept/split of function nit.

Reply:
Good suggestion, I implemented this, but this changed the PR quite significantly. Therefore re-requesting review...